editor/quick_open.cpp

```diff
@@ -113,12 +113,13 @@ void EditorQuickOpen::_sbox_input(const Ref<InputEvent> &p_ie) {
 
 float EditorQuickOpen::_path_cmp(String search, String path) const {
 
+	int pos = -1;
 	if (search == path) {
 		return 1.2f;
+	} else if ((pos = path.rfindn(search)) != -1) {
```

Comment:
We don't use statements in `if` conditions, please change it to:
```
} else {
    int pos = path.rfindn(search);
    if (pos != -1) {
        return 1.1f + 0.09 / (path.length() - pos + 1);
    }
}
```
I think a comment explaining the logic of the weighing would be useful too.

Reply:
@akien-mga I have made the required changes. The logic of the code is when we enter a keyword for file searching we prioritise files over folder. The current system yields the whole directory of the file in which filename is actually in the last. So by getting the first occurrence of the keyword from last which is then mapped such that the score lies between 1.1f to 1.19f where the occurrence of substring on the right side of the string has lesser score than that on the left side because the folders exist on the left side while the files on the right.
Sir, I also had another problem which is I am not able to squash my commits. The last commit I made was few months back so it is difficult to track back to that commit and fixup it.